.github/workflows/iroha2-dev-pr.yml

```diff
@@ -92,94 +94,43 @@ jobs:
           mold --run cargo llvm-cov nextest
-          --no-fail-fast
-          --workspace --lib
           --no-default-features
-          --branch
-          --no-report
-      - name: Run unit tests (all features)
+          --branch --no-report
+      - name: Test with all features
         run: >
           mold --run cargo llvm-cov nextest
-          --no-fail-fast
-          --workspace --lib
           --all-features
-          --branch
-          --no-report
+          --branch --no-report
       - name: Generate lcov report
         run: cargo llvm-cov report --lcov --output-path lcov.info
       - name: Upload lcov report
         uses: actions/upload-artifact@v4
         with:
           name: report-coverage
           path: lcov.info
-
-  # include: iroha/tests/integration/
-  # exclude: iroha/tests/integration/extra_functional
-  integration:
-    runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-09-09
-    needs: build_wasm_samples
-    timeout-minutes: 30
-    steps:
-      - uses: actions/checkout@v4
-      - name: Download executor.wasm
-        uses: actions/download-artifact@v4
-        with:
-          name: executor.wasm
-          path: ${{ env.DOCKER_COMPOSE_PATH }}
-      - name: Download the rest of WASM samples
-        uses: actions/download-artifact@v4
-        with:
-          name: wasm_samples
-          path: ${{ env.WASM_SAMPLES_TARGET_DIR }}
-      - uses: taiki-e/install-action@nextest
-      - name: Run integration tests, with all features
-        run: >
-          mold --run cargo nextest run 
-          --all-features
-          --no-fail-fast 
-          --failure-output immediate-final 
-          -E 'package(iroha) and test(integration) and not test(extra_functional)'
-
-  # include: iroha/tests/integration/extra_functional
-  extra_functional:
-    runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-09-09
-    needs: build_wasm_samples
-    timeout-minutes: 60
-    steps:
-      - uses: actions/checkout@v4
-      - name: Download executor.wasm
-        uses: actions/download-artifact@v4
-        with:
-          name: executor.wasm
-          path: ${{ env.DOCKER_COMPOSE_PATH }}
-      - name: Download the rest of WASM samples
-        uses: actions/download-artifact@v4
+      - name: Upload test network artifacts
+        if: always()
+        uses: actions/upload-artifact@v4
         with:
-          name: wasm_samples
-          path: ${{ env.WASM_SAMPLES_TARGET_DIR }}
-      - uses: taiki-e/install-action@nextest
-      - name: Run integration tests, with all features
-        run: >
-          mold --run cargo nextest run 
-          --all-features
-          --no-fail-fast 
-          --failure-output final 
-          --test-threads 1
-          -E 'test(extra_functional)'
+          name: test_network_runs
+          path: ${{ env.TEST_NETWORK_TMP_DIR }}/irohad_test_network_*
```

Comment:
Peer logs aren't printed during tests, except for STDERR (but not always). They are too verbose and it is much more convenient to inspect them stored in separate files, for separate peers.

So, these artifacts contain each generated peer's configurations, logs, and even Kura storage. If some test fails, it's possible to fetch the artifacts, find a specific directory, and inspect what was going on with a specific peer.

Reply:
If artifact size is a concern, we can consider tweaking `retention-days` and `compression-level`

Reply:
Set retention days to 3. Not sure if compression level will help much, as it seems the most space is occupied by the binary blobs of `genesis.scale` with pre-optimised WASMs inside.

Reply:
Would it make sense to only upload test artifacts if tests fail?

Reply:
@0x009922 

We could assume developers should review cause of the failure in 1 day, but I'd not insist on this.

Rather, `run-*-genesis.scale` for each run of each test looks redundant to me. I think only few tests are modifying the default test genesis. So each run could record some reference to genesis

Reply:
> Would it make sense to only upload test artifacts if tests fail?

Makes sense. 

> Rather, `run-*-genesis.scale` for each run of each test looks redundant to me. I think only few tests are modifying the default test genesis. So each run could record some reference to genesis

There is only one `genesis.scale` for each peer, it doesn't vary between runs. Plus, there are very few tests that restart peers. But anyways, while I totally agree that duplication of genesis is sub-optimal, implementing optimisation with a reference seems to be a premature optimisation to me.

~~Let me set the max compression level - maybe it will be smart-enough to de-duplicate `genesis.scale`?~~ I just remembered that each genesis contains trusted peers encoded, which is a unique value each time, so it's no use.